tally_ho/apps/tally/static/js/barcode_verify.js

```diff
@@ -1,27 +1,68 @@
 function barcodes_match(empty_message, length_message, mismatch_message) {
-    var barcode = document.getElementById('id_barcode');
-    var barcode_copy = document.getElementById('id_barcode_copy');
+    const barcode = document.getElementById('id_barcode');
+    const barcode_copy = document.getElementById('id_barcode_copy');
+    const barcode_scan_entry = document.getElementById('barcode_scan_entry').hidden;
+    if (barcode_scan_entry){
+        if (barcode.value == barcode_copy.value) {
+            barcode_copy.parentNode.setAttribute('class', 'has-success');
+            return true;
+        }
+        barcode_copy.parentNode.setAttribute('class', 'has-error');
 
-    if (barcode.value == barcode_copy.value) {
-        barcode_copy.parentNode.setAttribute('class', 'has-success');
-        return true;
-    }
-    barcode_copy.parentNode.setAttribute('class', 'has-error');
-
-    if (barcode.value == "" || barcode_copy.value == "") {
-        barcode.parentNode.setAttribute('class', 'has-error');
-        alert(empty_message);
-    } else if (barcode.value.length < 1) {
-        barcode.parentNode.setAttribute('class', 'has-error');
-        alert(length_message);
+        if (barcode.value == "" || barcode_copy.value == "") {
+            barcode.parentNode.setAttribute('class', 'has-error');
+            alert(empty_message);
+        } else if (barcode.value.length < 1) {
+            barcode.parentNode.setAttribute('class', 'has-error');
+            alert(length_message);
+        }
+        else {
+            alert(mismatch_message);
+        }
     }
-    else {
-        alert(mismatch_message);
+
+    return;
+}
+
+let barcode_number = '';
+
+const barcode_scanner = (barcode_input_field) => {
+    if (barcode_input_field) {
+        barcode_input_field.addEventListener('keydown', function (e) {
+            e.preventDefault();
+            let textInput = e.key
+            if (textInput.length === 1 || (e.keyCode === 13)) {
```

Comment:
Could you explain why these are split into separate `if` statements instead of combined,

```javascript
if ((textInput.length === 1 || e.keyCode === 13) && !isNaN(parseInt(textInput) && e.code === 'KeyA')
```

Reply:
Nice catch, Yeah this needs refactoring. Infact i don't need to check for `textInput.length`  since `!isNaN(parseInt(textInput)` ensure it's always an int and plus `e.code === 'KeyA'` will make the length one and a number.

I have re-written this to:
```
if (!(isNaN(parseInt(textInput))) && e.code === 'KeyA') {
    barcode_number = barcode_number + textInput;
} else if (barcode_number && (e.keyCode === 13)) {
       barcode_input_field.value = barcode_number;
       barcode_input_field.readOnly = true;
}
```